hkmc2/shared/src/main/scala/hkmc2/codegen/Lowering.scala

```diff
@@ -388,6 +388,11 @@ class Lowering(lowerHandlers: Bool, stackLimit: Option[Int])(using TL, Raise, St
               )
             pat match
               case Pattern.Lit(lit) => mkMatch(Case.Lit(lit) -> go(tail, topLevel = false))
+              case Pattern.ClassLike(cls: ClassSymbol, _trm, _args0, _refined)
+                  // Do not elaborate `_trm` when the `cls` is virtual.
+                  if Elaborator.ctx.Builtins.virtualClasses contains cls =>
+                // `Value.Arr(Nil)` is a dummy result and will not be used.
```

Comment:
Why pick this weird value as the dummy? Why will it not be used? What's doing on exactly, here? A concrete example could help.

Reply:
Some classes (e.g. `Int`) from `Prelude` does not exist at runtime. If we do lowering on `trm`, then `JSBuilder` will generate selection that crashes.

`Case.Cls` requires `path: Path` as the second parameter. Since the selection is not available, I just passed a dummy value here.

It's not used because if `JSBuilder` found the class symbol does not exist at runtime, then `JSBuilder` does not use the second parameter.

Reply:
Then make the dummy value something like `globalThis.Predef.unreachable` (and make that a getter that throws an internal error). And add the explanation you commented here as documentation in the actual code. How else do you expect someone will understand this part of the code? More over, in the doc, please include the string `[invariant:0]` and add the same `[invariant:0]` string in a comment on the corresponding `JSBuilder` part that discards the path. Because you're introducing a non-local invariant here that we need to be able to track! (Not to mention other backends will need to be careful about this as well.)